CI/deploy-mudlet-for-windows.sh

```diff
@@ -0,0 +1,331 @@
+  echo "FOLDER_TO_UPLOAD=$uploadDir" >> "$GITHUB_ENV"
+  echo "UPLOAD_FILENAME=$uploadFilename" >> "$GITHUB_ENV"
+}
+
+
+# Remove specific file types from the directory
+rm ./*.cpp ./*.o
+
+
+# Check if GITHUB_REPO_TAG is "false" and PublicTestBuild is not true
+if [[ "$GITHUB_REPO_TAG" == "false" ]] && [[ "$PublicTestBuild" == false ]]; then
+  echo "=== Creating a snapshot build ==="
+  mv "$PACKAGE_DIR/mudlet.exe" "Mudlet.exe"
+  
+  # Create a zip file using 7z
+  7z a "Mudlet-$VERSION$MUDLET_VERSION_BUILD-$BUILD_COMMIT-windows-$BUILD_BITNESS.zip" "$PACKAGE_DIR/*"
+  
+  # Define the upload filename
+  uploadFilename="Mudlet-$VERSION$MUDLET_VERSION_BUILD-$BUILD_COMMIT-windows-$BUILD_BITNESS.zip"
+  
+  # Move the zip file to the upload directory (assumed to be previously defined as a function or script)
+  moveToUploadDir "$uploadFilename"
+else
+
+  # Check if it's a Public Test Build
+  if [[ "$PublicTestBuild" == "true" ]]; then
+
+    # Get the commit date of the last commit
+    COMMIT_DATE=$(git show -s --format="%cs")
+    # Get yesterday's date in the same format
+    YESTERDAY_DATE=$(date --date="yesterday" +%Y-%m-%d)
+
+    if [[ "$COMMIT_DATE" < "$YESTERDAY_DATE" ]]; then
+      echo "=== No new commits, aborting public test build generation ==="
+      exit 3
+    fi
+
+    echo "=== Creating a public test build ==="
+    # Squirrel uses Start menu name from the binary, renaming it
+    mv "$PACKAGE_DIR/mudlet.exe" "$PACKAGE_DIR/Mudlet PTB.exe"
+    # ensure sha part always starts with a character due to a known issue
+    VersionAndSha="${VERSION}-ptb-${BUILD_COMMIT}"
+
+  else
+    echo "=== Creating a release build ==="
+    mv "$PACKAEG_DIR/mudlet.exe" "$PACKAGE_DIR/Mudlet.exe"
+    VersionAndSha="$VERSION"
+  fi
+
+  echo "=== Cloning installer project ==="
+  git clone https://github.com/Mudlet/installers.git "$GITHUB_WORKSPACE/installers"
+  cd "$GITHUB_WORKSPACE/installers/windows" || exit 1
+
+  echo "=== Installing Squirrel for Windows ==="
+  nuget install squirrel.windows -ExcludeVersion
+
+  echo "=== Setting up directories ==="
+  SQUIRRELWIN="$GITHUB_WORKSPACE/squirrel-packaging-prep"
+  SQUIRRELWINBIN="$SQUIRRELWIN/lib/net45/"
+
+  if [[ ! -d "$SQUIRRELWINBIN" ]]; then
+    mkdir -p "$SQUIRRELWINBIN"
+  fi
+
+  echo "=== Moving things to where Squirrel expects them ==="
+  mv "$PACKAGE_DIR/"* "$SQUIRRELWINBIN"
+
+  # Set the path to the nuspec file
+  NuSpec="$GITHUB_WORKSPACE/installers/windows/mudlet.nuspec"
+  echo "=== Creating Nuget package ==="
+
+  # Check if this is a Public Test Build
+  if [[ "$PublicTestBuild" == "true" ]]; then
+    # Allow public test builds to be installed side by side with the release builds by renaming the app
+    # No dots in the <id>: Guidelines by Squirrel
+    sed -i 's/<id>Mudlet<\/id>/<id>Mudlet-PublicTestBuild<\/id>/' "$NuSpec"
+    sed -i 's/<title>Mudlet<\/title>/<title>Mudlet (Public Test Build)<\/title>/' "$NuSpec"
```

Comment:
🚨 The names used here MUST be different for the 64-Bit builds (but the ID must still be a valid one - which rules out some obvious candidates) - see:
* https://github.com/Squirrel/Squirrel.Windows/blob/develop/docs/using/nuget-package-metadata.md
* https://github.com/Squirrel/Squirrel.Windows/blob/master/docs/using/naming.md

basically it looked as though `-` is used as a separator (but we seem to be getting away with it in the PTBs, nor `.` or spaces and I've found the id cannot **end** in numbers or use any sort of brace or bracket either. When I was trying something out in the past I had not changed the 32-bit ones from what **was already in use** but changed the 64-bit ones to use `Mudlet_64_` for the id for the release builds and `Mudlet_64_-PublicTestBuild` for the PTBs.